### PR TITLE
Remove dependency on eval

### PR DESCRIPTION
**Pull request by buu700**
I added a dependency on [math-expression-evaluator](https://github.com/redhivesoftware/math-expression-evaluator), which as far as I can tell should handle every expected input that's currently being passed in to eval.

### CODE REVIEW

**Comment by buu700** [comment]
Looks like the tests are failing because they aren't configured to work with dependencies/`require`? Not sure how you guys want to handle all that stuff, but in any case this is more of a suggestion than a real code contribution (my change to the JS was one line).

Tests aside, it seems to be working as expected in my project that uses @angular/animations.

**Comment by buu700** [comment]
Caught an edge case regression: repeatedly mousing in and out of an element with an @angular/material mdTooltip will at some point throw `Mismatched interpolation arguments 0:100`.

**Comment by ankitbug94** [comment]
Hi, I am creator of mexp. As far as I know, mexp.eval returns a Number with precision upto 15 digits. May you please tell me in which case it is returning a string?

**Comment by buu700** [comment]
It looks like it's consistently returning a string in both a node repl and my browser console:

screen shot 2017-04-28 at 4 06 11 am

**Comment by ankitbug94** [comment]
I have fixed this. Now it is returning numbers in version 1.2.17

**Comment by buu700** [comment]
Nice, thanks! I'll update this pr to remove `parseFloat` and bump up the mexp version.

File: src/dimension-handler.js [hunk 1]
```diff
@@ -14,6 +14,8 @@
 
 (function(scope, testing) {
 
+  var mexp = require('math-expression-evaluator');
```
Comment (dstockwell) [inline src/dimension-handler.js]: This isn't nodejs code, `require` wont work here.

File: src/dimension-handler.js [hunk 2]
```diff
@@ -54,7 +56,7 @@
       return;
 
     for (var unit in matchedUnits) {
-      var result = eval(string.replace(new RegExp('U' + unit, 'g'), '').replace(new RegExp(taggedUnitRegExp, 'g'), '*0'));
+      var result = mexp.eval(string.replace(new RegExp('U' + unit, 'g'), '').replace(new RegExp(taggedUnitRegExp, 'g'), '*0'));
```
Comment (dstockwell) [inline src/dimension-handler.js]: The man reason `eval` is used here is to keep the code size small. How large is this dependency? Further, if we're going to replace this, I think we would prefer to write the replacement inline rather than adding dependencies to this library.

File: src/dimension-handler.js (hunk 1 above)
Comment (buu700) [inline src/dimension-handler.js]: I wasn't sure how you guys would want to handle dependencies, since this would be the first one. Is there a solution that would be preferred over `require` with a module bundler?

One option that comes to mind is something like `typeof require !== 'undefined' ? require ('math-expression-evaluator').eval : eval`, which would accommodate both users like me who can't use `eval` for CSP compliance and users who want the size advantage you mentioned of not including another dependency (either by loading web-animations-js with a script tag or by configuring math-expression-evaluator as an undefined external in their module bundler).

File: src/dimension-handler.js (hunk 2 above)
Comment (buu700) [inline src/dimension-handler.js]: Looks like it's 11.7 kb.

Comment (ewilligers) [inline src/dimension-handler.js]: https://github.com/web-animations/web-animations-js/pull/151 implements calc expression inline and adds 449 bytes to web-animations-next-lite.min.js. No need to add an 11.7 kb dependency.

Comment (buu700) [inline src/dimension-handler.js]: Nice, that's great news!
